src/Http/Routing/src/Builder/EndpointRouteBuilderExtensions.cs

```diff
@@ -186,41 +181,21 @@ public static RouteGroupBuilder MapGroup(this IEndpointRouteBuilder endpoints, R
         this IEndpointRouteBuilder endpoints,
         RoutePattern pattern,
         RequestDelegate requestDelegate)
+    {
+        return Map(endpoints, pattern, requestDelegate, httpMethods: null);
+    }
+
+    private static IEndpointConventionBuilder Map(
+        this IEndpointRouteBuilder endpoints,
+        RoutePattern pattern,
+        RequestDelegate requestDelegate,
+        IEnumerable<string>? httpMethods)
     {
         ArgumentNullException.ThrowIfNull(endpoints);
         ArgumentNullException.ThrowIfNull(pattern);
         ArgumentNullException.ThrowIfNull(requestDelegate);
 
-        const int defaultOrder = 0;
-
-        var builder = new RouteEndpointBuilder(
-            requestDelegate,
-            pattern,
-            defaultOrder)
-        {
-            DisplayName = pattern.RawText ?? pattern.DebuggerToString(),
-        };
-
-        // Add delegate attributes as metadata
-        var attributes = requestDelegate.Method.GetCustomAttributes();
```

Comment:
I may have missed it but where are we populating metadata from attributes now?

Reply:
`RouteEndpointDataSource` now takes care of that bit since this removed the simpler `ModelEndpointDataSource` which would allow group conventions to unintentionally override endpoint-specific metadata like attributes.

Of course, when conventions are aware of groups, they might want to actually see this metadata as you pointed out #42750. The silver lining here is that these endpoints were never eligible for `.WithOpenApi()` treatment to begin with due to the lack of `MethodInfo` in the metadata, and this much remains unchanged so we don't make that issue any worse. And we're going to fix that issue hopefully anyway.